### ISSUE TITLE
How to retrieve RestClient attributes in a ClientHttpRequestInterceptor?

### ISSUE THREAD

**Issue by giovannitomasicchio** [spring-projects/spring-framework#31625]
**Affects:** Spring Framework 6.1.0-RC2

---

The new RestClient allows you to set attributes but it seems that they are not saved in the ClientHttpRequest so you can't retrieve them in a ClientHttpRequestInterceptor.

 So what are the RestClient attributes for?

**Comment by chrislhardin** [comment]
So I actually used the attributes but wasn't aware it was not working. What is the alternative to putting attributes onto the request in this? uriVariables is a HashMap that I was passing in. I don't see another way to add the request attributes other than calling url(u -> u.build(uriVariables)) but I think that is to build the actual url from its constituent components. I just want to pass the url in its entirety and then pass some variable onto it. I could expand them on the url String before I make the call I guess. 

`restClient.method(method).uri(url)
				.attributes(a -> Optional.ofNullable(uriVariables).ifPresent(v ->
				 a.putAll(uriVariables)))
				.headers(httpHeaders -> httpHeaders.addAll(headers)).retrieve().toEntity(responseType);`

**Comment by chrislhardin** [comment]
yeah I saw that but I was missing the url part so I ended up with this. I haven't tested it yet. 

`restClient.method(method).uri(url, uriBuilder -> uriBuilder.queryParams(uriVariables).build())
				.headers(httpHeaders -> httpHeaders.addAll(headers)).retrieve().toEntity(responseType);`

**Comment by KrishnaST** [comment]
@poutsma Why prefer ThreadLocals when a clean API can be made available to pass attributes?
